src/Graph.ts

```diff
@@ -74,6 +75,72 @@ function isAnExit(board: Board, x: number, y: number) {
   );
 }
 
+export function positionToCoordinate(position: string): {
+  x: number;
+  y: number;
+} {
+  if (position.length > 3) {
+    return { x: -1, y: -1 };
+  }
+
+  const positionSplit = position.split("");
+
+  const x = ALPHABET.indexOf(positionSplit[0]);
+  const y = parseInt(positionSplit[1] + positionSplit[2]);
+
+  if (!x || x === -1 || !y) {
+    return { x: -1, y: -1 };
+  }
+
+  return { x, y };
+}
+
+export function moveMarbleInDirection(
+  graph: Graph,
+  marbleCoordinate: { x: number; y: number },
+  direction: string
+): Graph {
+  if (!marbleCoordinate || !graph || !direction) {
+    return {
+      nodes: {},
+      edges: [],
+    };
+  }
+
+  let currentNode = graph.nodes[`${marbleCoordinate.x},${marbleCoordinate.y}`];
+
+  if (currentNode.value === 0) {
+    return graph;
+  }
+
+  const nodes = [currentNode];
+  while (true) {
```

Comment:
infinite loop ? how do you get off this loop? It will be better to add a condition to stop instead of break

Reply:
it is ok for the moment